qa/tasks/cephadm.py

```diff
@@ -757,6 +757,51 @@ def ceph_rgw(ctx, config):
 
     yield
 
+
+@contextlib.contextmanager
+def ceph_iscsi(ctx, config):
+    """
+    Deploy iSCSIs
+    """
+    cluster_name = config['cluster']
+    fsid = ctx.ceph[cluster_name].fsid
+
+    nodes = []
+    daemons = {}
+    for remote, roles in ctx.cluster.remotes.items():
+        for role in [r for r in roles
+                    if teuthology.is_type('iscsi', cluster_name)(r)]:
+            c_, _, id_ = teuthology.split_role(role)
+            log.info('Adding %s on %s' % (role, remote.shortname))
+            nodes.append(remote.shortname + '=' + id_)
```

Comment:
When I use the same (or similar) command that's being generated (`ceph orch apply iscsi iscsi user password --placement '1;ironic-moliver=iscsi.a'`) it fails as the container is cleaned up as an orphan daemon.

So maybe that's happening. If I don't set a name, it's fine for me. Although I'm not sure if this will work in teuthology and I'm not sure if it needs the `id_` for anything else.
But just to check we could attempt to change this to:
`nodes.append(remote.shortname)`

Reply:
Oh though then we may not know the darmon name down in register daemons.

So the qa software way have a fit.

Might need to go look at why the damn daemon name is so problematic and why orphan daemons is potentially raising it's ugly head again.

Reply:
Hm. thought I fixed this my chaining the name to `iscsi.iscsi.a`

Reply:
> Hm. thought I fixed this my chaining the name to `iscsi.iscsi.a`

Right: 

from http://pulpito.ceph.com/swagner-2020-06-24_11:51:16-rados:cephadm-wip-swagner2-testing-2020-06-24-1115-distro-basic-smithi/5175577/

```
14-11ea-a06d-001a4aab830c -- ceph orch apply iscsi iscsi user password --placement '1;smithi184=iscsi.a'
```
then `orch ps`:

```
NAME              HOST       STATUS          REFRESHED  AGE   VERSION               IMAGE NAME                                                          IMAGE ID      CONTAINER ID
...
iscsi.iscsi.a     smithi184  unknown         18s ago    98s   <unknown>             quay.ceph.io/ceph-ci/ceph:f62be9f20bb0fda935cc0d6b707f0066d129e03f  <unknown>     <unknown>
...
```

which means that the daemon_name is `iscsi.iscsi.a`. Note that I had to fix cephadm in order to prevent it from cleaning up orphaned daemons:

https://github.com/ceph/ceph/pull/35543/files#diff-40c4f610f9f7b519e2c2053ffdc27d92R1306-R1309